src/pysdmx/model/dataflow.py

```diff
@@ -112,6 +118,28 @@ class Component(Struct, frozen=True, omit_defaults=True):
     local_codes: Union[Codelist, Hierarchy, None] = None
     attachment_level: Optional[str] = None
     array_def: Optional[ArrayBoundaries] = None
+    attribute_relationship: Optional[Dict[str, "Component"]] = None
```

Comment:
I don't understand the purpose of this new property. 

There is already the `attachment_level` property. This property expects one of the following:

- `D` for a dataset-level attribute
- `O` for an observation-level attribute
- And a **comma-separated list of dimension IDs** for all 3 other cases (group-level and series-level attributes, as well as attributes attached to one or more dimensions that represent neither a group nor a series).

So, considering a series with 3 dimensions (dim1, dim2 and dim3) and an attribute attached to that series, that attribute would have an attachment-level of  `dim1,dim2,dim3`, i.e. a string easily splitable at the commas, to get the list of dimensions.

Is this not what this new `attribute_relationship` is expected to hold? If so, I guess we don't need both?

Reply:
Checking again the documentation on attachment level you are right, it is redundant.

Two things to consider:
- We do also have a dictionary in dataset with a dictionary of attributes, for those at Dataset level.
- Need to change as well the data writers as well, will include in this PR

Reply:
Changed the attachment level and data writers for correct attribute relationship.

Reply:
Modified as well the validations before writing to ensure we get only the attributes that are not related to the Dataset

Reply:
@sosna Ready to review this again